webapp/src/app.tsx

```diff
@@ -2874,6 +2874,18 @@ export class ProjectView
         tc.toggleHint(showFullText);
     }
 
+    ///////////////////////////////////////////////////////////
+    ////////////     User alert/notifications     /////////////
+    ///////////////////////////////////////////////////////////
+
+    pokeUserActivity() {
+        if (!!this.state.tutorialOptions && !!this.state.tutorialOptions.tutorial) {
+            // animate tutorial hint after some time of user inactivity
+            let tc = this.refs["tutorialcard"] as tutorial.TutorialCard;
+            if (tc) tc.startPokeUserActivity();
```

Comment:
IMO, we should have a centralized "tutorial engine" that is in charge of receiving pokes, debouncing and changing the correct state to generate animation -- we don't want timers scattered in the UI. So move the timer from the tutorial card into a "hint engine" (might as well start a new file to keep things tidy) and centralize operations there.